Kernel/Syscalls/sched.cpp

```diff
@@ -68,8 +68,13 @@ ErrorOr<FlatPtr> Process::sys$scheduler_set_parameters(Userspace<Syscall::SC_sch
     if (!credentials->is_superuser() && credentials->euid() != peer_credentials->uid() && credentials->uid() != peer_credentials->uid())
         return EPERM;
 
-    // FIXME: Only sets priority for main thread of process if mode == PROCESS
     peer->set_priority((u32)parameters.parameters.sched_priority);
+    if (parameters.mode == Syscall::SchedulerParametersMode::Process) {
+        peer->process().for_each_thread([&](auto& thread) {
+            thread.set_priority((u32)parameters.parameters.sched_priority);
+        });
+    }
+
```

Comment:
This could probably use a comment that explains why we are doing this (e.g. "POSIX says that process scheduling parameters have precedence over thread scheduling parameters. We don't track them separately, so overwrite the thread scheduling settings manually for now.").